crates/router/src/connector/noon/transformers.rs

```diff
@@ -330,17 +336,37 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
                 },
             });
 
-        let (subscription, tokenize_c_c) =
-            match item.request.setup_future_usage.is_some().then_some((
-                NoonSubscriptionData {
-                    subscription_type: NoonSubscriptionType::Unscheduled,
-                    name: name.clone(),
-                },
-                true,
-            )) {
-                Some((a, b)) => (Some(a), Some(b)),
-                None => (None, None),
-            };
+        let (subscription, tokenize_c_c) = match item.request.setup_future_usage.is_some() {
```

Comment:
`match` statement can be refactored from `setup_future_usage` as we can use `setup_mandate_details` like used in stripe.